docs/proposals/env-vars.md

```diff
@@ -0,0 +1,316 @@
+
+The next major dependency for Shipwright, Tekton, has a vast enough array of scenarios it addresses that it provides a 
+direct line to the Kubernetes Container API in their API.  As as result, environment variable fields from `Steps`/`Containers`
+are directly accessible from Tekton API.
+
+Specifically [Tekton TaskRuns](https://github.com/tektoncd/pipeline/blob/main/docs/taskruns.md) reference
+[Tekton Steps](https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#defining-steps), and those embed the
+[core Kubernets Containers type](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#container-v1-core).
+
+There are also a set of features in Tekton around setting values for environment variables in a dynamic fashion.
+
+To complete the chain of dependencies, Shipwright translates its API objects ultimately into Tekton TaskRuns.
+
+While the list of scenarios Shipwright goes after is a subset of what Tekton goes after, as the [Motivation](#motivation)
+section will explain, manipulation of environment variable settings on the resulting k8s Container objects exits
+for Shipwright.
+
+How to best allow Shipwright users to manage in a first class way from our API environment variables that are ultimately set on k8s 
+Containers is what this enhancement proposal will address.
+
+
+## Motivation
+
+The various image management tools Shipwright encapsulates with 'BuildStrategies' all allow for tuning of their behavior
+via environment variables.  The names and supported values of those variables vary between tools, and sometimes 
+between different versions of those tools.  So a lot of variability we need to account for.
+
+The same goes for source code management tools, tools for retrieving content via socket connections, both of which are 
+used in Shipwright.
+
+So supporting our users via our API and related features to manage environment variables for those tools makes a lot of
+sense.
+
+### Goals
+
+- Shipwright API needs to articulate a first class way for specifying those environment variables approach for both 
+  administrators and individual developers (i.e. authors/owners of the BuildStrategies vs. authors owners of the Builds 
+  and BuildRuns) to specify those environment variables.
+- Use of the existing features Tekton provides to enrich our users' management of environment variables needs to be 
+  evaluated and exploited for code reuse.
+- Maintain the currently Shipwright approach of treating Tekton as an implementation detail, and try to not leak Tekton
+  API into our API.
+
+### Non-Goals
+
+- Any needed change to integrate Shipwright with Tekton parameters is handled by the [current 'Parameterize Build Strategies' EP](https://github.com/shipwright-io/build/pull/697)
+  or follow up work that stems from that EP.  Until then, only static content of any environment variable value related 
+  fields is supported.  In other words, the `$(params...)` syntax seen in Tekton today.
+
```

Comment:
To make it clear, it may make sense to add a statement that this EP is not about environment variables that will be set in the final container image.

Reply:
sure that distinction has value ... I'll add